tiledb/sm/query/domain_buffer.h

```diff
@@ -106,6 +106,89 @@ class DomainBufferDataRef : public detail::DomainBuffersTypes,
   DomainBufferDataRef(const Domain& domain) = delete;
 };
 
+/**
+ * A container to store data for a single coordinate value across all
+ * dimensions.
+ */
+class SingleCoord {
+ public:
+  /**
+   * Default constructor is prohibited.
+   */
+  SingleCoord() = delete;
+
+  /**
+   * Constructor
+   *
+   * @param dim_num the number of dimensions
+   */
+  SingleCoord(unsigned dim_num)
+      : coords_(dim_num)
+      , qb_(dim_num)
+      , sizes_(dim_num)
+      , single_offset_(0)
+      , single_offset_size_(sizeof(uint64_t)) {
+  }
+
+  /**
+   * Store the values of the coordinate to the passed in data ref.
+   *
+   * @param schema Array schema.
+   * @param coord Data ref to the coordinate.
+   */
+  void set(const ArraySchema& schema, const DomainBufferDataRef& coord) {
```

Comment:
Can we move this to the constructor and make this class C.41 compliant?

Reply:
@ihnorton it is possible, but that introduces the need to pass in a valid coordinate at creation of the global write state. When there is not a valid entry yet. I used the first cell for now, or we can convert last_cell_coords_ to a pointer, thoughts?

Reply:
There is no "last written coordinate" until a write has actually occurred, so how about `optional<SingleCoord>`? (and throw if there's an attempt to use it before initialization)